### ISSUE TITLE
Assertion in bn_div_words

### ISSUE THREAD

**Issue by paradust7** [openssl/openssl#18045]
Trying to use curl for HTTPS in emscripten, every request failed with "wrong signature length" inside int_rsa_verify().

After enabling debugging, there was an assertion in bn_div_words. I was able to fix this by adding SIXTY_FOUR_BIT and -DBN_DIV2W to the system configuration (linux-generic64) to bypass this version of the function entirely. After this change, curl with https functions correctly.

The assertion with context:

```
/* Divide h,l by d and return the result. */
/* I need to test this some more :-( */
BN_ULONG bn_div_words(BN_ULONG h, BN_ULONG l, BN_ULONG d)
{
    BN_ULONG dh, dl, q, ret = 0, th, tl, t;
    int i, count = 2;

    if (d == 0)
        return BN_MASK2;

    i = BN_num_bits_word(d);
    assert((i == BN_BITS2) || (h <= (BN_ULONG)1 << i));  <<<<<<<<<<<<<<<<
```

Built from source openssl-1.1.1n.tar.gz

Built using:

```
emconfigure ./Configure --debug linux-generic64 \
  no-asm \
  no-engine \
  no-hw \
  no-weak-ssl-ciphers \
  no-dtls \
  no-shared \
  no-dso \
  -DPEDANTIC \
  --prefix="$INSTALL_DIR"

sed -i 's|^CROSS_COMPILE.*$|CROSS_COMPILE=|g' Makefile

emmake make -j4 build_generated libssl.a libcrypto.a
```

```
RuntimeError: Aborted(Assertion failed: (i == BN_BITS2) || (h <= (BN_ULONG)1 << i), at: crypto/bn/bn_asm.c,216,bn_div_words)
    at abort (eval at importScripts (minimal.worker.js:39:16), <anonymous>:1384:10)
    at ___assert_fail (eval at importScripts (minimal.worker.js:39:16), <anonymous>:1991:2)
    at bn_div_words (<anonymous>:wasm-function[2994]:0x27305d)
    at bn_div_fixed_top (<anonymous>:wasm-function[3002]:0x29bf19)
    at BN_div (<anonymous>:wasm-function[3001]:0x29b4d6)
    at BN_nnmod (<anonymous>:wasm-function[3008]:0x29daab)
    at ec_GFp_simple_group_set_curve (<anonymous>:wasm-function[3742]:0x3361e1)
    at ec_GFp_nist_group_set_curve (<anonymous>:wasm-function[3730]:0x330fc8)
    at EC_GROUP_set_curve (<anonymous>:wasm-function[3659]:0x31fc78)
    at EC_GROUP_new_curve_GFp (<anonymous>:wasm-function[3591]:0x318ae4)
    at ec_group_new_from_data (<anonymous>:wasm-function[3527]:0x3090a9)
    at EC_GROUP_new_by_curve_name (<anonymous>:wasm-function[3526]:0x308a31)
    at eckey_type2param (<anonymous>:wasm-function[3459]:0x2fc58f)
    at eckey_pub_decode (<anonymous>:wasm-function[3458]:0x2fc1c2)
    at x509_pubkey_decode (<anonymous>:wasm-function[6136]:0x55c2a6)
    at pubkey_cb (<anonymous>:wasm-function[6142]:0x55cc2d)
    at asn1_item_embed_d2i (<anonymous>:wasm-function[2694]:0x246735)
    at asn1_template_noexp_d2i (<anonymous>:wasm-function[2699]:0x249413)
    at asn1_template_ex_d2i (<anonymous>:wasm-function[2695]:0x246ed9)
    at asn1_item_embed_d2i (<anonymous>:wasm-function[2694]:0x2461e4)
    at asn1_template_noexp_d2i (<anonymous>:wasm-function[2699]:0x249413)
    at asn1_template_ex_d2i (<anonymous>:wasm-function[2695]:0x246ed9)
    at asn1_item_embed_d2i (<anonymous>:wasm-function[2694]:0x2461e4)
    at ASN1_item_ex_d2i (<anonymous>:wasm-function[2693]:0x24464f)
    at ASN1_item_d2i (<anonymous>:wasm-function[2692]:0x2444d8)
    at d2i_X509 (<anonymous>:wasm-function[6144]:0x55cdb5)
    at PEM_X509_INFO_read_bio (<anonymous>:wasm-function[5123]:0x46841c)
    at X509_load_cert_crl_file (<anonymous>:wasm-function[5766]:0x535245)
    at by_file_ctrl (<anonymous>:wasm-function[5770]:0x5358d3)
    at X509_LOOKUP_ctrl (<anonymous>:wasm-function[5823]:0x5390d8)
    at X509_STORE_load_locations (<anonymous>:wasm-function[5802]:0x53819a)
    at SSL_CTX_load_verify_locations (<anonymous>:wasm-function[1773]:0x195c66)
    at ossl_connect_step1 (<anonymous>:wasm-function[1195]:0x12ce7b)
    at ossl_connect_common (<anonymous>:wasm-function[1182]:0x128903)
    at ossl_connect_nonblocking (<anonymous>:wasm-function[1183]:0x1291d1)
    at Curl_ssl_connect_nonblocking (<anonymous>:wasm-function[1245]:0x13c855)
    at https_connecting (<anonymous>:wasm-function[453]:0x55215)
    at Curl_http_connect (<anonymous>:wasm-function[451]:0x54e62)
    at multi_runsingle (<anonymous>:wasm-function[530]:0x6d49f)
    at curl_multi_perform (<anonymous>:wasm-function[528]:0x6c323)
    at easy_transfer (<anonymous>:wasm-function[239]:0x21c1b)
    at easy_perform (<anonymous>:wasm-function[237]:0x218c5)
    at curl_easy_perform (<anonymous>:wasm-function[236]:0x214f1)
    at __original_main (<anonymous>:wasm-function[93]:0x55db)
    at main (<anonymous>:wasm-function[94]:0x567c)
```

**Comment by faramir-dev** [comment]
@paradust7 I have created issue: https://github.com/openssl/openssl/issues/18750 for adding of official support of Wasm to OpenSSL and we will discuss it. Wasm is more and more important these days. At this moment, if everything works for you, then your solution is probably the best.

**Comment by nhorman** [comment]
Marking as inactive, will be closed at the end of 3.4 dev barring further input.
